pkg/liqoctl/inband/cluster.go

```diff
@@ -833,13 +860,18 @@ func (c *Cluster) DisablePeering(ctx context.Context, remoteClusterID *discovery
 		return nil
 	}
 
+	// Do not proceed if the peering is not in-band.
+	if fc.Spec.PeeringType != discoveryv1alpha1.PeeringTypeInBand {
+		return fmt.Errorf("the peering type towards remote cluster %q is %s, expected %s",
+			remName, fc.Spec.PeeringType, discoveryv1alpha1.PeeringTypeInBand)
+	}
+
 	// Set outgoing peering to no.
-	if _, err := controllerutil.CreateOrPatch(ctx, c.local.CRClient, fc, func() error {
+	if _, err := controllerutil.CreateOrUpdate(ctx, c.local.CRClient, fc, func() error {
```

Comment:
```suggestion
	if _, err = controllerutil.CreateOrUpdate(ctx, c.local.CRClient, fc, func() error {
```

We can avoid the re-definition of `err`